### PR TITLE
Gppkg update versioning

### PR DESCRIPTION
**Pull request by Chibin**
We were using string comparison for our version checks and did not account for two digit numbers.
For example:
comparing '5.0.10.0' to '5.0.5.0' will return 5.0.5.0 being greater

Switching to magic number comparison, fixing this issue.

### CODE REVIEW

File: gpMgmt/bin/gppylib/operations/package.py [hunk 1]
```diff
@@ -399,13 +404,26 @@ def _get_gpdb_version(self):
...
+            Get the version of the current GPDB magic number
+            Returns an int consisting of the major and minor release version
+        '''
+        logger.debug('_convert_to_magic_number_version')
+
+        ver_list = gpversion_obj.version
+
+        # This is retreived from our configure.in file that generates
+        # GP_VERSION_NUM
+        magic_num = "%d%02d%02d" % (ver_list[0], ver_list[1],
```
Comment (larham) [inline gpMgmt/bin/gppylib/operations/package.py]: I'm not clear on what this comment means.  Is "ver_list" derived from configure.in?  What is GP_VERSION_NUM, is that related to ver_list?

File: gpMgmt/bin/gppylib/test/unit/test_unit_package.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    def _is_requires_orca_logged(self, gppkg_name, log_messages):
+        return ('Greenplum Database requires orca version of '
+                '%s' % gppkg_name in log_messages)
+
+    @patch('gppylib.operations.package.GpVersion',
+           return_value=Mock(version=[4, 3, 10, 0]))
+    @patch('gppylib.operations.package.logger',
+           return_value=Mock(spec=['log', 'info', 'debug', 'error']))
+    def test__execute_reports_incompatability(self,
```
Comment (larham) [inline gpMgmt/bin/gppylib/test/unit/test_unit_package.py]: since logger is mocked out for each and every test, consider mocking it out in a "setUp()" method common to all tests (GpTestCase and its superclasses have the correct signature; see elsewhere in unit tests to find examples)

**Comment by Chibin** [comment]
> I'm not clear on what this comment means. Is "ver_list" derived from configure.in? What is GP_VERSION_NUM, is that related to ver_list?

Sorry, I needed to be clearer. When I meant "this", I actually meant the way magic_num/GP_VERSION_NUM is generated.

**Comment by cjcjameson** [comment]
@mktal @orhankislal @njayaram2 @iyerr3 see for code and tests regarding gppkg version number changes

**Comment by Chibin** [comment]
> since logger is mocked out for each and every test, consider mocking it out in a "setUp()" method common to all tests (GpTestCase and its superclasses have the correct signature; see elsewhere in unit tests to find examples)

Good point. I considered it, but wasn't sure if it should be done now as there is only 3 tests.
Let me do that.

**Comment by Chibin** [comment]
Addressed PR comments

**Comment by Chibin** [comment]
Thanks for the reviews! Feel free to review this PR further to see if it requires more changes.
